platform/server-bundle/src/main/java/org/motechproject/server/web/controller/AvailabilityController.java

```diff
@@ -0,0 +1,102 @@
+package org.motechproject.server.web.controller;
+
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Controller class responsible for determining available UI tabs for modules where
+ * UI tabs availability depends on different user permissions.
+ */
+@Controller
+public class AvailabilityController {
```

Comment:
I believe this controller was supposed to determine available tabs, based on the data in ModuleRegistrationData, not hardcode settings for each module. What if external module wants to use it? They have to edit platform code?